File: integrations/src/main/java/services/voice/provider/VoiceProvider.java
package services.voice.provider;

import framework.ThirdPartyProvider;
import framework.config.BusinessContext;
import framework.core.data.ThirdPartyConfig;
import services.voice.dto.VoiceCallRequest;
import services.voice.dto.VoiceCallResponse;
import services.voice.provider.exotel.data.ExotelConfiguration;

public interface VoiceProvider extends ThirdPartyProvider<ExotelConfiguration> {
    VoiceCallResponse makeCall(
            VoiceCallRequest request,
            ThirdPartyConfig config,
            BusinessContext businessContext
    );
    
    VoiceCallResponse getCallStatus(
            String callSid,
            ThirdPartyConfig config,
            BusinessContext businessContext
    );
}

File: integrations/src/main/kotlin/services/voice/provider/VoiceProvider.kt
package services.voice.provider

import framework.ThirdPartyProvider
import framework.config.BusinessContext
import framework.core.data.ThirdPartyConfig
import services.voice.dto.VoiceCallRequest
import services.voice.dto.VoiceCallResponse
import services.voice.provider.exotel.data.ExotelConfiguration

interface VoiceProvider : ThirdPartyProvider<ExotelConfiguration> {
    fun makeCall(
        request: VoiceCallRequest,
        config: ThirdPartyConfig,
        businessContext: BusinessContext
    ): VoiceCallResponse
    fun getCallStatus(callSid: String, config: ThirdPartyConfig, businessContext: BusinessContext): VoiceCallResponse
}
